services/web/server/src/simcore_service_webserver/login/_registration.py

```diff
@@ -74,47 +87,78 @@ def ensure_enum(cls, v):
-        #  can be overtaken by this new registration
-        #
-        if user["status"] == CONFIRMATION_PENDING:
-            _confirmation = await db.get_confirmation(
-                filter_dict={
-                    "user": user,
-                    "action": ConfirmationAction.REGISTRATION.value,
-                }
-            )
-            drop_previous_registration = not _confirmation or is_confirmation_expired(
-                cfg, _confirmation
-            )
-            if drop_previous_registration:
-                if not _confirmation:
-                    await db.delete_user(user=user)
-                else:
-                    await db.delete_confirmation_and_user(
-                        user=user, confirmation=_confirmation
-                    )
+        user_status = UserStatus(user["status"])
+        match user_status:
 
-                _logger.warning(
-                    "Re-registration of %s with expired %s"
-                    "Deleting user and proceeding to a new registration",
-                    f"{user=}",
-                    f"{_confirmation=}",
+            case UserStatus.ACTIVE:
+                await _raise_if_registered_in_product(
+                    app, user_email=user["email"], product=current_product
                 )
-                return
 
-        raise web.HTTPConflict(
-            reason=MSG_EMAIL_EXISTS, content_type=MIMETYPE_APPLICATION_JSON
-        )
+            case UserStatus.CONFIRMATION_PENDING:
+                # An account already registered with this email
+                #
+                #  RULE 'drop_previous_registration': any unconfirmed account w/o confirmation or
+                #  w/ an expired confirmation will get deleted and its account (i.e. email)
+                #  can be overtaken by this new registration
+                #
+                # resend confirmation ??
+                _confirmation = await db.get_confirmation(
+                    filter_dict={
+                        "user": user,
+                        "action": ConfirmationAction.REGISTRATION.value,
+                    }
+                )
+                drop_previous_registration = (
+                    not _confirmation or is_confirmation_expired(cfg, _confirmation)
+                )
+                if drop_previous_registration:
+                    if not _confirmation:
+                        await db.delete_user(user=user)
+                    else:
+                        await db.delete_confirmation_and_user(
+                            user=user, confirmation=_confirmation
+                        )
+
+                    _logger.warning(
+                        "Re-registration of %s with expired %s"
+                        "Deleting user and proceeding to a new registration",
+                        f"{user=}",
+                        f"{_confirmation=}",
+                    )
```

Comment:
Q: what happens if:
- the user already exists and has projects in product A
- an invitation to product B is send and expires
- a second invitation to product C is sent


Will the user loose all their data and projects?

Reply:
Could we have a test for this?

Reply:
No, the user will not loose all their data and projects. Some reasons:
1. the rule you refer to only applies to the "old invitations" 
   - old invitations do not know about products only emails
   - old invitations are deprecated and not produced anymore. Unlikely that is used
1. `db.delete` used here would fail because of foreign constraints. A user cannot be deleted if it has data/projects. This is one of the reasons why removing accounts is not automatically done.